src/main/scala/mesosphere/marathon/metrics/current/reporters/DataDogAPIReporter.scala

```diff
@@ -0,0 +1,151 @@
+package mesosphere.marathon
+package metrics.current.reporters
+
+import java.net.InetAddress
+import java.time.Clock
+import java.util.concurrent.TimeUnit
+
+import akka.actor.{Actor, Props}
+import akka.http.scaladsl.Http
+import akka.http.scaladsl.model.{HttpEntity, HttpMethods, HttpRequest, HttpResponse, MediaTypes, StatusCodes}
+import akka.stream.{ActorMaterializer, ActorMaterializerSettings}
+import akka.util.ByteString
+import com.codahale.metrics.{Counter, Gauge, Histogram, Metered, MetricRegistry, Snapshot, Timer}
+import com.typesafe.scalalogging.StrictLogging
+import mesosphere.marathon.metrics.MetricsConf
+
+import scala.collection.JavaConverters._
+import scala.concurrent.duration.FiniteDuration
+
+class DataDogAPIReporter(metricsConf: MetricsConf, registry: MetricRegistry) extends Actor with StrictLogging {
+  private val apiKey = metricsConf.metricsDataDogApiKey()
+  private val apiUrl = s"https://app.datadoghq.com/api/v1/series?api_key=$apiKey"
+  private val transmissionIntervalMs = metricsConf.metricsDataDogTransmissionIntervalMs()
+  private val transmissionIntervalS = transmissionIntervalMs / 1000
+  private val http = Http(context.system)
+  private val host = InetAddress.getLocalHost.getHostName
+
+  private case object Tick
+
+  import akka.pattern.pipe
+  import scala.concurrent.ExecutionContext.Implicits.global
+
+  private implicit val materializer: ActorMaterializer = ActorMaterializer(ActorMaterializerSettings(context.system))
+
+  override def preStart(): Unit = {
+    self ! Tick
+  }
+
+  override def receive: Receive = {
+    case Tick =>
+      report()
+    case HttpResponse(code, _, entity, _) =>
```

Comment:
I think we must call `entity.discardBytes()` in case we don't consume it, otherwise we will trigger backpressure.

Reply:
Great catch. Thank you!